pkg/tracegen/distributed.go

```diff
@@ -19,22 +19,25 @@ package tracegen
 
 import (
 	"context"
+	"fmt"
 
-	apmhttp "go.elastic.co/apm/module/apmhttp/v2"
-	"go.uber.org/zap"
+	"go.elastic.co/apm/v2"
 )
 
 // SendDistributedTrace sends events generated by both APM Go Agent and OTEL library and
 // link them with the same traceID so that they are linked and can be shown in the same trace view
-func SendDistributedTrace(ctx context.Context, cfg *Config, logger *zap.SugaredLogger) error {
-	cfg.TraceID = newRandomTraceID()
-	txCtx, err := IndexIntakeV2Trace(ctx, *cfg, logger)
+func SendDistributedTrace(ctx context.Context, cfg Config) error {
```

Comment:
This function should be the one calling `cfg.Validate()` it shouldn't be left out to the client to do that.